File: app/src/main/java/com/ocram/qichwadic/domain/repository/DictionaryRepository.java
package com.ocram.qichwadic.domain.repository;

import com.ocram.qichwadic.domain.model.Definition;
import com.ocram.qichwadic.domain.model.Dictionary;

import java.util.List;

import io.reactivex.Flowable;

public interface DictionaryRepository {

    Flowable<List<Dictionary>> getDictionaries();

    Flowable<List<Dictionary>> checkSavedDictionaries();

    Flowable<Integer> getSavedDictionariesTotal();

    Flowable<List<Definition>> getDefinitionsByDictionary(int dictionaryId);

    void saveDictionaryAndDefinitions(Dictionary dictionary, List<Definition> definitions);

    void removeDictionary(int id);
}

File: app/src/main/java/com/ocram/qichwadic/features/dictionaries/domain/DictionaryRepository.kt
package com.ocram.qichwadic.features.dictionaries.domain

import com.ocram.qichwadic.core.domain.model.DefinitionModel
import com.ocram.qichwadic.core.domain.model.DictionaryModel
import kotlinx.coroutines.flow.Flow

interface DictionaryRepository {

    suspend fun getCloudDictionaries(): List<DictionaryModel>

    suspend fun saveNewDictionaries(dictionaries: List<DictionaryModel>)

    suspend fun getSavedDictionaries(): Flow<List<DictionaryModel>>

    suspend fun getDefinitionsByDictionary(dictionaryId: Int): List<DefinitionModel>

    suspend fun saveDefinitions(definitions: List<DefinitionModel>)

    suspend fun removeDictionary(id: Int): Int
}